ui/app/AppLayouts/Profile/popups/TokenListPopup.qml

```diff
@@ -0,0 +1,285 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQml.Models 2.15
+import QtQuick.Layouts 1.15
+
+import StatusQ.Core 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Components 0.1
+import StatusQ.Popups.Dialog 0.1
+import StatusQ.Core.Theme 0.1
+
+import utils 1.0
+
+import shared.panels 1.0
+
+StatusDialog {
+    id: root
+
+    /*required*/ property string sourceName
+    /*required*/ property string sourceImage
+    /*required*/ property string sourceUrl
+    /*required*/ property int sourceUpdatedAt
+    /*required*/ property string sourceVersion
+    /*required*/ property int tokensCount
+    /*required*/ property var tokensListModel // Expected roles: name, symbol, image, chainName, explorerUrl
```

Comment:
Why not make them _really_ required? :)

Reply:
Yeah, it's a bit unclear for the reader, maybe it's worth to mention it's done like that bc popup is used via Instantiator

Reply:
Ok, commenting `required` keyword because of limitations in specific use-case is not elegant...
The alternative is to keep required keywords but wrap popup when used as a delegate:

```
    Instantiator {
        model: SortFilterProxyModel {
            sourceModel: sourcesOfTokensModel

            filters: ValueFilter {
                id: keyFilter

                roleName: "key"
                value : ""
            }
        }

        delegate: QtObject {
            id: delegate

            required property string name
            required property string image
            required property string source
            required property int updatedAt
            required property string version
            required property int tokensCount

            readonly property TokenListPopup popup: TokenListPopup {
                parent: root

                sourceName: delegate.name
                sourceImage: delegate.image
                sourceUrl: delegate.source
                sourceVersion: delegate.version
                sourceUpdatedAt: delegate.updatedAt
                tokensCount: delegate.tokensCount

                tokensListModel: SortFilterProxyModel {
                    sourceModel: root.tokensProxyModel

                    // Filter by source
                    filters: ExpressionFilter {
                        expression: (model.jsArraySources).includes(keyFilter.value)
                    }
                }

                onLinkClicked: logs.logEvent("TokenListPopup::onLinkClicked --> " + link)
                onClosed: keyFilter.value = ""
                Component.onCompleted: open()
            }
        }
    }
```